src/components/resource-selector/media-library/media-library-item-modal.js

```diff
@@ -55,63 +129,151 @@ function MediaLibaryMetadataModal({
+          <div className="MediaLibaryItemModal-footerButtons MediaLibaryItemModal-footerButtons--default">
+            <Button type="primary" onClick={onClose}>{t('common:ok')}</Button>
+          </div>
+        </div>
+      );
+      break;
+    case MEDIA_LIBRARY_METADATA_MODAL_MODE.update:
+      isDialogClosable = false;
+      dialogTitle = t('common:edit');
+      dialogContent = (
+        <div className="MediaLibaryItemModal">
+          <div className="MediaLibaryItemModal-displayName">
+            {mediaLibraryItem.displayName}
+          </div>
+          <div className="MediaLibaryItemModal-splitView">
             <ResourceDetails url={mediaLibraryItem.url} size={mediaLibraryItem.size} previewOnly />
-            <MediaLibraryMetadataForm form={form} file={mediaLibraryItem} useOptimizeImage={false} onFinish={handleFinish} />
+            <MediaLibraryMetadataForm form={form} file={mediaLibraryItem} useOptimizeImage={false} onFinish={handleUpdateItemFinish} />
           </div>
-        )}
-        {!!mediaLibraryItem && mode === MEDIA_LIBRARY_METADATA_MODAL_MODE.preview && (
-          <div className="MediaLibaryMetadataModal-url">
-            <ResourceUrl url={mediaLibraryItem.url} />
+        </div>
+      );
+      dialogFooter = (
+        <div className="MediaLibaryItemModal-footer">
+          <div className="MediaLibaryItemModal-footerButtons MediaLibaryItemModal-footerButtons--default">
+            <Button onClick={onClose}>{t('common:cancel')}</Button>
+            <Button type="primary" loading={isSaving} onClick={handleSaveClick}>{t('common:save')}</Button>
+          </div>
+        </div>
+      );
+      break;
+    case MEDIA_LIBRARY_METADATA_MODAL_MODE.create:
+      if (isEditingImage) {
+        isDialogClosable = false;
+        dialogTitle = t('common:create');
+        dialogContent = (
+          <div className="MediaLibaryItemModal">
+            <div className="MediaLibaryItemModal-imageEditor">
+              <ImageEditor
+                file={fileInfo.file}
+                editorRef={imageEditorRef}
+                onCrop={handleImageEditorCrop}
+                />
+            </div>
+          </div>
+        );
+        dialogFooter = (
+          <div className="MediaLibaryItemModal-footer">
+            <div className="MediaLibaryItemModal-footerButtons MediaLibaryItemModal-footerButtons--extra">
+              <Button onClick={handleImageEditorBackClick}>{t('common:back')}</Button>
+            </div>
+            <div className="MediaLibaryItemModal-footerButtons MediaLibaryItemModal-footerButtons--default">
+              <Button onClick={onClose}>{t('common:cancel')}</Button>
+              <Button type="primary" loading={isSaving} disabled={!isEditedImageDirty} onClick={handleApplyImageEditorChanges}>{t('common:apply')}</Button>
+            </div>
+          </div>
+        );
+      } else {
+        isDialogClosable = false;
+        dialogTitle = t('common:create');
```

Comment:
Tiny thing, these two lines can go above the `if` statement, since they are repeated. Cause I initially thought the we need different setup based in case.

Reply:
True, done!